src/main/java/de/tum/in/www1/artemis/domain/TextExercise.java

```diff
@@ -17,21 +15,20 @@
  */
 @Entity
 @DiscriminatorValue(value = "T")
+@JsonTypeName("text")
 @SecondaryTable(name = "text_exercise_details")
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
 public class TextExercise extends Exercise {
 
+    // used to distinguish the type when used in collections (e.g. SearchResultPageDTO --> resultsOnPage)
+    public String getType() {
+        return "text";
+    }
+
     @Column(name = "example_solution")
     private String exampleSolution;
 
-    @OneToMany(mappedBy = "exercise", cascade = CascadeType.REMOVE)
-    @JsonIgnore
-    private List<TextCluster> clusters;
-
-    @ManyToOne
-    @JoinColumn(table = "text_exercise_details")
-    @JsonIgnore
-    private TextAssessmentKnowledge knowledge;
+    // TODO: we should add a OneToMany to TextCluster with delete cascade
```

Comment:
Did you create a Github issue for that?

Reply:
No, we should not add one for every TODO